File: SportsSideKick/app/src/main/java/base/app/util/ui/ImageLoader.java
package base.app.util.ui;

import android.content.Context;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.nostra13.universalimageloader.core.DisplayImageOptions;

public class ImageLoader {

    private static ImageLoader instance;

    public static ImageLoader getInstance() {
        if (instance == null) {
            instance = new ImageLoader();
        }
        return instance;
    }

    public void displayImage(String uri, ImageView imageView) {
        displayImage(uri, imageView, null);
    }

    public void displayImage(String uri, ImageView imageView, DisplayImageOptions options) {
        Context ctx = imageView.getContext();
        Glide.with(ctx).load(uri).into(imageView);
    }
}

File: SportsSideKick/app/src/main/java/base/app/util/ui/ImageLoader.kt
package base.app.util.ui

import android.widget.ImageView
import base.app.R
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

fun ImageView.show(uri: Any?, error: Int? = null) {
    ImageLoader.displayImage(uri, this, error)
}

object ImageLoader {

    @JvmStatic
    fun displayRoundImage(uri: Any?, view: ImageView?) {
        displayImage(uri, view, R.drawable.blank_profile_rounded, true)
    }

    @JvmStatic
    @JvmOverloads
    fun displayImage(uri: Any?, view: ImageView?, error: Int? = null, isRound: Boolean = false) {
        if (uri != null) {
            if (view != null && view.context != null) {
                try {
                    Glide.with(view.context)
                            .load(uri)
                            .apply(optionsWith(error, isRound))
                            .into(view)
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
        } else {
            if (error != null) {
                displayImage(error, view)
            }
        }
    }

    private fun optionsWith(errorRes: Int?, isRound: Boolean): RequestOptions {
        var options = RequestOptions()
        if (errorRes != null) {
            options = options.error(errorRes)
        }
        if (isRound) {
            options = options.circleCrop()
        }
        return options
    }
}
